brokerapi/brokers/account_managers/sql_account_manager.go

```diff
@@ -39,10 +40,16 @@ type SqlAccountManager struct {
 }
 
 // inserts a new user into the database and creates new ssl certs
-func (sam *SqlAccountManager) CreateCredentials(instanceID string, bindingID string, details models.BindDetails, instance models.ServiceInstanceDetails) (models.ServiceBindingCredentials, error) {
+func (sam *SqlAccountManager) CreateCredentials(instanceID string, bindingID string, details brokerapi.BindDetails, instance models.ServiceInstanceDetails) (models.ServiceBindingCredentials, error) {
+
+	bindParameters := map[string]interface{}{}
+	if err := json.Unmarshal(details.RawParameters, &bindParameters); err != nil {
+		return models.ServiceBindingCredentials{}, err
+	}
+
 	var err error
-	username, usernameOk := details.Parameters["username"].(string)
-	password, passwordOk := details.Parameters["password"].(string)
+	username, usernameOk := bindParameters["username"].(string)
+	password, passwordOk := bindParameters["password"].(string)
 
 	if !passwordOk || !usernameOk {
```

Comment:
Even though it's less concise, it might make sense to split these into two separate error messages for clarity to save users some time while debugging.